File: 2022 czerwiec/INF.04-02-22.06-SG/Aplikacja Konsolowa/C++/main.cpp
#include <iostream>

using namespace std;

class Osoba
{
private:
    int id;
    string name;

public:
    static int count;

    Osoba()
    {
        id = 0;
        name = "";
        count++;
    }

    Osoba(int newId, string newName)
    {
        id = newId;
        name = newName;
        count++;
    }

    Osoba(const Osoba &otherPerson)
    {
        id = otherPerson.id;
        name = otherPerson.name;
        count++;
    }

    void showName(string otherName)
    {
        if (!name.empty())
        {
            cout << "Cześć " << otherName << ", mam na imię " << name << endl;
        }
        else
        {
            cout << "Brak danych!" << endl;
        }
    }
};

int Osoba::count = 0;

int main()
{
    cout << "Liczba zarejestrowanych osób to: " << Osoba::count << endl;

    Osoba osoba1;

    int id;
    string name;

    cout << "Podaj id osoby: ";
    cin >> id;
    cout << "Podaj imię osoby: ";
    cin >> name;

    Osoba osoba2(id, name);
    Osoba osoba3 = osoba2;

    osoba1.showName("Jan");
    osoba2.showName("Jan");
    osoba3.showName("Jan");

    cout << "Liczba zarejestrowanych osób to: " << Osoba::count << endl;

    return 0;
}

File: 2022 czerwiec/INF.04-02-22.06-SG/Aplikacja Konsolowa/Python/main.py
class Osoba:
    count = 0
    
    def __init__(self, id = 1, name = ""):
        self.__id = id
        self.__name = name
        Osoba.count += 1
        
    def copy(self, otherPerson):
        self.__id = otherPerson.__id
        self.__name = otherPerson.__name
        return
        
    def showName(self, othername):
        if (self.__name != ""):
            print("Cześć ", othername, ", mam na imię ", self.__name, sep="")
        else:
            print("Brak danych!")
        
print("Liczba zarejestrowanych osób to: ", Osoba.count);        
        
osoba1 = Osoba()

id = input("Podaj id osoby: ")
name = input("Podaj imię osoby: ")

osoba2 = Osoba(id, name)
osoba3 = Osoba()
osoba3.copy(osoba2)

osoba1.showName("Jan")
osoba2.showName("Jan")
osoba3.showName("Jan")

print("Liczba zarejestrowanych osób to: ", Osoba.count);  
